spec/numberformat.html

```diff
@@ -182,28 +182,34 @@ <h1>FormatNumericToString ( _intlObject_, _x_ )</h1>
       </emu-alg>
     </emu-clause>
 
-    <emu-clause id="sec-partitionnumberpattern" aoid="PartitionNumberPattern">
-      <h1>PartitionNumberPattern ( _numberFormat_, _x_ )</h1>
-
-      <p>
-        The PartitionNumberPattern abstract operation is called with arguments _numberFormat_ (which must be an object initialized as a NumberFormat) and _x_ (which must be a Number or BigInt value), interprets _x_ as a numeric value, and creates the corresponding parts according to the effective locale and the formatting options of _numberFormat_. The following steps are taken:
-      </p>
+    <emu-clause id="sec-partitionnumberpattern" type="abstract operation">
+      <h1>
+        PartitionNumberPattern (
+          _numberFormat_: an object initialized as a NumberFormat,
+          _x_: a Number or a BigInt,
+        )
+      </h1>
+      <dl class="header">
+        <dt>description</dt>
+        <dd>It creates the parts representing the mathematical value of _x_ according to the effective locale and the formatting options of _numberFormat_.</dd>
+      </dl>
 
       <emu-alg>
         1. Let _exponent_ be 0.
         1. If _x_ is *NaN*, then
           1. Let _n_ be an implementation- and locale-dependent (ILD) String value indicating the *NaN* value.
-        1. Else if _x_ is *+&infin;*, then
+        1. Else if _x_ is *+&infin;*<sub>𝔽</sub>, then
           1. Let _n_ be an ILD String value indicating positive infinity.
-        1. Else if _x_ is *-&infin;*, then
+        1. Else if _x_ is *-&infin;*<sub>𝔽</sub>, then
           1. Let _n_ be an ILD String value indicating negative infinity.
         1. Else,
-          1. If _numberFormat_.[[Style]] is *"percent"*, let _x_ be 100 × _x_.
+          1. Set _x_ to ℝ(_x_).
+          1. If _numberFormat_.[[Style]] is *"percent"*, set _x_ to 100 × _x_.
           1. Let _exponent_ be ComputeExponent(_numberFormat_, _x_).
-          1. Let _x_ be _x_ × 10<sup>-_exponent_</sup>.
+          1. Set _x_ to _x_ × 10<sup>-_exponent_</sup>.
           1. Let _formatNumberResult_ be FormatNumericToString(_numberFormat_, _x_).
```

Comment:
@sffc There is an unrelated problem here: FormatNumericToString expects a Number or BigInt value, while _x_ is a mathematical value that AFAICT might suffer output-affecting lossiness if converted.

And similar problems in later steps (in which whether or not _x_ is a mathematical value is dependent upon whether or not the original input was finite!):
1. GetNumberFormatPattern performs incoherent comparisons on its input, `_x_ is 0 or _x_ &gt; 0` is only valid for mathematical `_x_` while `_x_ is *NaN*` can only be true if `_x_` is a Number.
2. Likewise, PartitionNotationSubPattern seems to expect a mathematical _x_ but then compares it to infinities and `*NaN*`.

I think the best fix would be updating FormatNumericToString/GetNumberFormatPattern/PartitionNotationSubPattern/etc. to accept new parameters allowing them to identify special non-mathematical Number values such as negative zero and infinities and NaN.